test/integration/router_container_test.rb

```diff
@@ -13,4 +13,17 @@
     response = @app.get('/back/home', lint: true)
     response.body.must_equal 'back'
   end
+
+  it 'matches against host' do
+    @router_container = Hanami::Router.new(scheme: 'https', host: 'hanami.test', port: 443) do
+      mount Front::App, at: '/front', host: 'www.hanami.test'
```

Comment:
The main problem is that this `host` parameter actually matches the exact host. It's not clearly documented in the http_router gem but I found it looking at the tests. Specifying the host parameter will actually call the [`Node#add_host`](https://github.com/joshbuddy/http_router/blob/master/lib/http_router/node.rb#L37) method that relies on the [`Node::Host`](https://github.com/joshbuddy/http_router/blob/master/lib/http_router/node/host.rb) class. This class doesn't override the `to_code` method (defined on [`Node::AbstractRequestNode`](https://github.com/joshbuddy/http_router/blob/master/lib/http_router/node/abstract_request_node.rb#L20))  that, by default, seems to execute a simple equality comparison.

I was thinking to do a PR on http_router gem in the future, so that the `host` param can work against a regexp.